#! /bin/sh /usr/share/dpatch/dpatch-run
## 30_modules_gui_wxwindows.dpatch by  <crimsun@fungus.sh.nu>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: wxwidgets 2.6 gui updates from 0.8.3 branch

@DPATCH@

diff -uN -x '*.svn*' -x 'Makefile*' /home/crimsun/porting/todo/vlc-0.8.2/modules/gui/wxwindows/dialogs.cpp 0.8.3/modules/gui/wxwindows/dialogs.cpp
--- /home/crimsun/porting/todo/vlc-0.8.2/modules/gui/wxwindows/dialogs.cpp	2005-06-25 08:43:09.000000000 -0500
+++ 0.8.3/modules/gui/wxwindows/dialogs.cpp	2005-08-02 19:27:52.058966800 -0500
@@ -2,7 +2,7 @@
  * dialogs.cpp : wxWindows plugin for vlc
  *****************************************************************************
  * Copyright (C) 2000-2004 VideoLAN
- * $Id: dialogs.cpp 11019 2005-05-15 13:20:55Z ipkiss $
+ * $Id: dialogs.cpp 11692 2005-07-10 12:12:34Z zorglub $
  *
  * Authors: Gildas Bazin <gbazin@videolan.org>
  *
@@ -158,11 +158,6 @@
     /* Create the messages dialog so it can begin storing logs */
     p_messages_dialog = new Messages( p_intf, p_parent ? p_parent : this );
 
-    /* Check if user wants to show the bookmarks dialog by default */
-    wxCommandEvent dummy_event;
-    if( config_GetInt( p_intf, "wxwin-bookmarks" ) )
-        OnBookmarks( dummy_event );
-
     /* Intercept all menu events in our custom event handler */
     PushEventHandler( new MenuEvtHandler( p_intf, NULL ) );
 
diff -uN -x '*.svn*' -x 'Makefile*' /home/crimsun/porting/todo/vlc-0.8.2/modules/gui/wxwindows/interface.cpp 0.8.3/modules/gui/wxwindows/interface.cpp
--- /home/crimsun/porting/todo/vlc-0.8.2/modules/gui/wxwindows/interface.cpp	2005-06-25 08:43:09.000000000 -0500
+++ 0.8.3/modules/gui/wxwindows/interface.cpp	2005-08-02 19:27:52.041969384 -0500
@@ -2,7 +2,7 @@
  * interface.cpp : wxWindows plugin for vlc
  *****************************************************************************
  * Copyright (C) 2000-2005 VideoLAN
- * $Id: interface.cpp 11292 2005-06-04 20:49:12Z dionoea $
+ * $Id: interface.cpp 11692 2005-07-10 12:12:34Z zorglub $
  *
  * Authors: Gildas Bazin <gbazin@videolan.org>
  *
@@ -225,7 +225,7 @@
     /* Initializations */
     p_intf = _p_intf;
     i_old_playing_status = PAUSE_S;
-    b_extra = VLC_FALSE;
+    b_extra =  config_GetInt( p_intf, "wxwin-extended-gui" );
 //    b_undock = VLC_FALSE;
 
 
diff -uN -x '*.svn*' -x 'Makefile*' /home/crimsun/porting/todo/vlc-0.8.2/modules/gui/wxwindows/wizard.cpp 0.8.3/modules/gui/wxwindows/wizard.cpp
--- /home/crimsun/porting/todo/vlc-0.8.2/modules/gui/wxwindows/wizard.cpp	2005-06-25 08:43:09.000000000 -0500
+++ 0.8.3/modules/gui/wxwindows/wizard.cpp	2005-08-02 19:27:52.043969080 -0500
@@ -2,7 +2,7 @@
  * wizard.cpp : wxWindows plugin for vlc
  *****************************************************************************
  * Copyright (C) 2000-2004 VideoLAN
- * $Id: wizard.cpp 10641 2005-04-10 18:40:52Z zorglub $
+ * $Id: wizard.cpp 11774 2005-07-17 18:23:43Z courmisch $
  *
  * Authors: Clément Stenac <zorglub@videolan.org>
  *
@@ -1085,7 +1085,7 @@
         }
     }
     p_parent->SetStream( methods_array[i_method].psz_access ,
-                         (char *)address_txtctrl->GetValue().mb_str() );
+                         address_txtctrl->GetValue().mb_str() );
 
     /* Set the action for the muxer page */
     ((wizEncapPage*)GetNext())->SetAction( p_parent->GetAction() );
@@ -1469,7 +1469,8 @@
     this->i_to = i_to;
 }
 
-void WizardDialog::SetTranscode( char *vcodec, int vb, char *acodec,int ab)
+void WizardDialog::SetTranscode( char const *vcodec, int vb,
+                                 char const *acodec, int ab)
 {
     if( strcmp( vcodec, "dummy") )
     {
@@ -1483,18 +1484,18 @@
     this->ab = ab;
 }
 
-void WizardDialog::SetStream( char *method, char *address )
+void WizardDialog::SetStream( char const *method, char const *address )
 {
     this->method = strdup( method );
     this->address = strdup( address );
 }
 
-void WizardDialog::SetTranscodeOut( const char *address )
+void WizardDialog::SetTranscodeOut( char const *address )
 {
     this->address = strdup( address );
 }
 
-void WizardDialog::SetMux( char *mux )
+void WizardDialog::SetMux( char const *mux )
 {
     this->mux = strdup( mux );
 }
diff -uN -x '*.svn*' -x 'Makefile*' /home/crimsun/porting/todo/vlc-0.8.2/modules/gui/wxwindows/wxwindows.cpp 0.8.3/modules/gui/wxwindows/wxwindows.cpp
--- /home/crimsun/porting/todo/vlc-0.8.2/modules/gui/wxwindows/wxwindows.cpp	2005-06-25 08:43:09.000000000 -0500
+++ 0.8.3/modules/gui/wxwindows/wxwindows.cpp	2005-08-02 19:27:52.054967408 -0500
@@ -2,7 +2,7 @@
  * wxwindows.cpp : wxWindows plugin for vlc
  *****************************************************************************
  * Copyright (C) 2000-2005 VideoLAN
- * $Id: wxwindows.cpp 10818 2005-04-26 08:34:54Z fenrir $
+ * $Id: wxwindows.cpp 11692 2005-07-10 12:12:34Z zorglub $
  *
  * Authors: Gildas Bazin <gbazin@netcourrier.com>
  *
@@ -90,9 +90,8 @@
 #define EMBED_TEXT N_("Embed video in interface")
 #define EMBED_LONGTEXT N_("Embed the video inside the interface instead " \
     "of having it in a separate window.")
-#define BOOKMARKS_TEXT N_("Show bookmarks dialog")
-#define BOOKMARKS_LONGTEXT N_("Show bookmarks dialog when the interface " \
-    "starts.")
+#define EXTENDED_TEXT N_("Show extended GUI")
+#define EXTENDED_LONGTEXT N_("Show extended interface at startup." )
 #define TASKBAR_TEXT N_("Show taskbar entry")
 #define TASKBAR_LONGTEXT N_("Show taskbar entry")
 #define MINIMAL_TEXT N_("Minimal interface")
@@ -121,14 +120,14 @@
 
     add_bool( "wxwin-embed", 1, NULL,
               EMBED_TEXT, EMBED_LONGTEXT, VLC_FALSE );
-    add_bool( "wxwin-bookmarks", 0, NULL,
-              BOOKMARKS_TEXT, BOOKMARKS_LONGTEXT, VLC_FALSE );
     add_bool( "wxwin-taskbar", 1, NULL,
               TASKBAR_TEXT, TASKBAR_LONGTEXT, VLC_FALSE );
     add_bool( "wxwin-minimal", 0, NULL,
               MINIMAL_TEXT, MINIMAL_LONGTEXT, VLC_TRUE );
     add_bool( "wxwin-autosize", 1, NULL,
               SIZE_TO_VIDEO_TEXT, SIZE_TO_VIDEO_LONGTEXT, VLC_TRUE );
+    add_bool( "wxwin-extended-gui", VLC_FALSE, NULL,
+              EXTENDED_TEXT, EXTENDED_LONGTEXT, VLC_FALSE );
 #ifdef wxHAS_TASK_BAR_ICON
     add_bool( "wxwin-systray", 0, NULL,
               SYSTRAY_TEXT, SYSTRAY_LONGTEXT, VLC_FALSE );
diff -uN -x '*.svn*' -x 'Makefile*' /home/crimsun/porting/todo/vlc-0.8.2/modules/gui/wxwindows/wxwindows.h 0.8.3/modules/gui/wxwindows/wxwindows.h
--- /home/crimsun/porting/todo/vlc-0.8.2/modules/gui/wxwindows/wxwindows.h	2005-06-25 08:43:09.000000000 -0500
+++ 0.8.3/modules/gui/wxwindows/wxwindows.h	2005-08-02 19:27:52.056967104 -0500
@@ -2,7 +2,7 @@
  * wxwindows.h: private wxWindows interface description
  *****************************************************************************
  * Copyright (C) 1999-2005 VideoLAN
- * $Id: wxwindows.h 11236 2005-06-01 19:53:37Z courmisch $
+ * $Id: wxwindows.h 11774 2005-07-17 18:23:43Z courmisch $
  *
  * Authors: Gildas Bazin <gbazin@videolan.org>
  *
@@ -761,18 +761,18 @@
 {
 public:
     /* Constructor */
-    WizardDialog( intf_thread_t *p_intf, wxWindow *p_parent,char *, int, int );
+    WizardDialog( intf_thread_t *, wxWindow *p_parent, char *, int, int );
     virtual ~WizardDialog();
-    void SetTranscode( char *vcodec, int vb, char *acodec,int ab);
+    void SetTranscode( char const *vcodec, int vb, char const *acodec, int ab);
     void SetMrl( const char *mrl );
     void SetTTL( int i_ttl );
     void SetPartial( int, int );
-    void SetStream( char *method, char *address );
-    void SetTranscodeOut( const char *address );
+    void SetStream( char const *method, char const *address );
+    void SetTranscodeOut( char const *address );
     void SetAction( int i_action );
     int  GetAction();
     void SetSAP( bool b_enabled, const char *psz_name );
-    void SetMux( char *mux );
+    void SetMux( char const *mux );
     void Run();
     int i_action;
     char *method;
